### DISCUSSION TITLE
Beat fires tasks on start

### DISCUSSION THREAD

**Discussion by Distortedlogic** [celery/celery#7408] [Q&A] [answered]
I am using CeleryBeat for cron jobs. Sometimes when I start celery the tasks will fire on startup even tho its not the time I scheduled for them to fire. I cannot find any info on this issue anywhere, other than for RedBeat, which im not using. https://github.com/sibson/redbeat/issues/79

[a comment by an automated account is not shown]

**Comment by thedrow** [comment]
Sorry. This is not enough for us for this issue to be actionable.
We need you to fill all the required information in the template and provide us with a minimally reproducible test case.

**Comment by Distortedlogic** [comment]
Sorry, I just do not have time atm. I just created a work around and rolling with it. For a touch of info while I am here with a little time, I have a celerybeat schedule that fires a task every hour. When I startup celery it will fire this job even tho it is not on the hour. Just drop in ur own task function to reproduce.
![17-01-2022-07-19-52](https://user-images.githubusercontent.com/54854447/149768473-60306087-03de-4ad7-910e-cf9fb75aade5.png)
![17-01-2022-07-18-30](https://user-images.githubusercontent.com/54854447/149768324-e9c54431-ebbe-4327-bf1f-0a1b11184a28.png)
![17-01-2022-07-18-40](https://user-images.githubusercontent.com/54854447/149768352-aca7d53c-3a4d-4479-9373-d89ffafa46a4.png)

**Comment by Smixi** [comment]
Which version are you using ? 

I'm running celery version 5.2 with the same config as you (I put the same config with the same backend/broker), but I tried to make it run every minutes, then at a specific minutes, and I couldn't reproduce by stopping and starting celery beat. 


However of course, If i do the opposite, start beat and send a task every minutes, If  then start celery worker, multiple tasks will be received and started by this worker. This is because the scheduler  send the tasks to the broker, not to the worker. So maybe there is some tasks pending in your  queue ?

**Comment by Distortedlogic** [comment]
Thanks for the response!
`celery == 5.2.1` 

> So maybe there is some tasks pending in your queue ?

great suggestion, offhand I believe u r right and that I shoulda thought bout the broker. I will confirm once I get home but def think this is it.

**Comment by Distortedlogic** [comment]
@Smixi  Unfortunately this did not resolve. I cleared all my docker images, volumes etc. Then on the fresh build and start, Beat still fired my tasks at startup :(

**Comment by auvipy** [comment]
transferring this to a discussion

**Comment by iamr0b0tx** [comment]
I am having the same issue. whenever I deploy and the beat service reloads the periodic tasks are executed event when its not time

**Comment by ZinnurovArtur** [comment]
Hello everyone, same issue there. When i try to deploy docker containers with Celery, Celery beat and Redis. Celery beat tries to run tasks on the startup which should not be right. Purge not solving the problem.

**Comment by kjg531** [comment]
Same issue here. Does not fire everytime I restart.

**Comment by amin-prvn** [comment]
> I am using CeleryBeat for cron jobs. Sometimes when I start celery the tasks will fire on startup even tho its not the time I scheduled for them to fire. I cannot find any info on this issue anywhere, other than for RedBeat, which im not using. [sibson/redbeat#79](https://github.com/sibson/redbeat/issues/79)

I have this problem too, also on azure web app and google cloud run platform.
Do you find a solution for this?

**Reply by Distortedlogic** [reply]
Sorry, but no. As of current I do not have a solution nor has it been addressed/resolved by the maintainers.

**Reply by Distortedlogic** [reply]
Best I can think of is to have the job itself ensure its not executing at the wrong time

**Comment by ltrcx** [comment]
Any update on the issue?

**Comment by thebalaa** [comment, marked as answer]
I was able to track this down and resolve in my case.

Watch out for the `celerybeat-schedule` file. I had it .gitignored but when I switched from mounted volumes to building containers it got picked up and put into my production deployment.

Deleting `celerybeat-schedule` file and rebuilding my containers resolved the issue of celery firing scheduled tasks on reboot.

Hope this helps someone else!

**Reply by JamesToohey** [reply]
Thanks so much for sharing this

**Reply by Distortedlogic** [reply]
idk if this will work for everyone but worked for me so i marked as the answer

**Comment by tsingwang** [comment]
You can refer here https://github.com/celery/celery/discussions/8076#discussioncomment-5056578
